packages/cli/src/commands/publish/submit.ts

```diff
@@ -42,19 +43,38 @@ export const publishSubmitCommand = async ({
     release: releaseDetails,
     solana_mobile_dapp_publisher_portal: solanaMobileDappPublisherPortalDetails,
   } = await getConfigFile();
+
   const sign = ((buf: Buffer) =>
     nacl.sign(buf, signer.secretKey)) as SignWithPublisherKeypair;
 
-  await publishSubmit(
-    { connection, sign },
-    {
-      appMintAddress: appMintAddress ?? appDetails.address,
-      releaseMintAddress: releaseMintAddress ?? releaseDetails.address,
-      publisherDetails,
-      solanaMobileDappPublisherPortalDetails,
-      compliesWithSolanaDappStorePolicies,
-      requestorIsAuthorized,
-    },
-    dryRun
-  );
+  const pubAddr = publisherDetails.address;
+  const appAddr = appMintAddress ?? appDetails.address;
+  const releaseAddr = releaseMintAddress ?? releaseDetails.address;
+
+  try {
+    const results = await connection.getMultipleAccountsInfo([
+      new PublicKey(pubAddr),
+      new PublicKey(appAddr),
+      new PublicKey(releaseAddr),
+    ]);
+
+    if (results?.length == 3) {
+      await publishSubmit(
+        { connection, sign },
+        {
+          appMintAddress: appAddr,
+          releaseMintAddress: releaseAddr,
+          publisherDetails,
+          solanaMobileDappPublisherPortalDetails,
+          compliesWithSolanaDappStorePolicies,
+          requestorIsAuthorized,
+        },
+        dryRun
+      );
+    } else {
+      throw new Error("");
+    }
+  } catch (e) {
```

Comment:
This catches the error you just threw above, and rethrows a new one, which will suppress the stack trace of the actual error. Why not move this message up to line 75 and let any other errors that come from `publishSubmit` propagate naturally?

Reply:
This will be updated for sure